compiler/optimizer/Dominators.hpp

```diff
@@ -55,30 +54,26 @@ class TR_Dominators
    int             dominates(TR::Block *block, TR::Block *other);
 
    TR::Compilation * comp()         { return _compilation; }
-   TR_Memory *      trMemory()      { return comp()->trMemory(); }
-   TR_StackMemory   trStackMemory() { return trMemory(); }
    bool trace() { return _trace; }
 
-   TR::deque<int32_t> _dfNumbers;
-
    protected:
 
-   typedef TR::SparseBitVector SparseBitVector;
+   typedef TR_BitVector BitVector;
 
    private:
 
    struct BBInfo
       {
-      BBInfo(TR::Allocator allocator) :
-         _bucket(allocator), _block(NULL), _parent(-1), _idom(-1), _ancestor(-1), _label(-1), _child(-1), _sdno(-1), _size(0)
+      explicit BBInfo(TR::Region &region) :
+         _bucket(region), _block(NULL), _parent(-1), _idom(-1), _ancestor(-1), _label(-1), _child(-1), _sdno(-1), _size(0)
          {}
       TR::Block     *_block;     // The block whose info this is
       int32_t       _parent;    // The parent in the depth-first spanning tree
       int32_t       _idom;      // The immediate dominator for this block
       int32_t       _ancestor;  // The ancestor in the forest
       int32_t       _label;     // The node in the ancestor chain with minimal
                                 // semidominator number
-      SparseBitVector _bucket;    // The blocks whose semidominator is this node
+      BitVector _bucket;    // The blocks whose semidominator is this node
```

Comment:
This looks like a place to expect a sparse bit vector to be an appropriate choice based on the thought that the set of "blocks whose semidominator is this node" is likely not a set that approaches the full set of blocks. However it may be that that is okay as a high level thought, but when one looks at the details of the sparse bit vector implementation it is possible that it takes more memory in some cases like reasonable sized (though not small) control flow graphs. So my question is : is there something inherent here that this set is'nt as sparse as I think it is OR is it that our sparse bit vector implementation does'nt quite work as well ?